### PR TITLE
Implement cell types editing, plotting, and prototypes for training

### PR DESCRIPTION
**Pull request by ykevu**
## What
This merges in all of the work that has been done on implementing the new cell types features, including importing, editing, adding, removing cell types as well as channel marker calculations, plotting, and in-browser training and prediction.
### Enable basic functionality for channel name editing and import + refine cell types controls (#393)
* Cell type colors now mix properly (for cells with multiple cell types as well as overlapping cells)
* Hovering over cells now show which cell type(s) that cell is labeled with
* Implement remove cells mode
* Added instructions for cell type labeling
* Deleting cells deletes them from cell types now
* Cell types are separately tracked by feature (segmentation)
* Color map bug fixes
* Channel names can now be edited
* Display controls visually tweaked
* Cell type opacity can be toggled and adjusted
* Channel name can be imported from OME.tiff metadata
### Implement channel expression calculations and plotting features + added linear opacity scaling (#394)
* Calculate mean and total channel expressions for an image and plot the values
* Clusters can be selected using plotly and added to specified cell types
* Other UI changes: linear label opacity scaling, zooming in with the mouse scroll up instead of out
* Cell grid removed.
### Add UMAP visualization, remove and cancel buttons, and other UI tweaks/fixes (#395)
* UMAP calculated for mean and total (some bugs associated)
* Scatter points turn white when selected
* Add/remove/cancel set of buttons added
* Minor UI tweaks and bug fixes
* Number of cells is now indicated in each cell type
* Toggle all button can toggle all cell types
* Scatter/histogram can be toggled with radio group
* Editing prompt position fixed
* Changed thickness of selected points on plot
* Fixed resizing to perform resizing by channel
* Fixed max iterations for cell type rendering
### Optimize multi-frame canvas rendering, implement prototype for in-browser training (#396)
* Added icons and made various small QOL and visual changes to the UI like plot titles
* Training/prediction tab added to prototype transfer learning feature
* Web worker memory bug "fixed" as the original raw array is no longer passed to the web worker, so larger files can be loaded now. This causes the issue of making it impossible to export the raw original though, so will be reverted 
* Relevant canvases now use a smaller cell matrix and use the actual list of cells on a given frame
* Training loss plot
* Training does not auto-predict
* Additional UI changes: hover tooltip persisting on edges, plotly settings, etc.
* Training tab moved to modal
* New state machine specifically for training created
* User can now specify percentage of training data to be used for training vs. validation (100% training not implemented)
* Validation curve added
* Confusion matrix calculated and displayed based on the validation data (not fully implemented for larger data)
### Misc. testing and code clean-up
* Fixed a toggle all bug
* Fixed a number of linting and formatting issues
* Added cypress test to test cell type controls
* CI changes/fixes

### CODE REVIEW

**Review by rossbar — APPROVED** [review]
Thanks for this @ykevu - getting this magnitude of code in is no small feat!
